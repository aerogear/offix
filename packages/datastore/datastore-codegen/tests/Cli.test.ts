import * as util from 'util';
import { mkdirSync, rmdirSync, readFileSync } from "fs";
import { join } from 'path';

beforeEach(() => mkdirSync("./output/"));
afterEach(() => rmdirSync("./output/", { recursive: true }));

test("generate", async () => {
    const exec = util.promisify(require('child_process').exec);
    const cliExec = join(__dirname, '../dist/index.js');
    const schemaPath = join(__dirname, './mock.graphql');

    await exec(`node ${cliExec} generate --schema ${schemaPath} --outputPath output/`);
    const schema = readFileSync("./output/schema/schema.json");
    const schemaIndex = readFileSync("./output/schema/index.ts");
    const config = readFileSync("./output/config.ts");

    expect(config).toBeDefined();
    expect(schema).toBeDefined();
    expect(schemaIndex).toBeDefined();
});
