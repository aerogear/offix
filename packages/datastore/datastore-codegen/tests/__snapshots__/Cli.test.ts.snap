// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate 1`] = `
"export interface Note {
    id: string;
    title: string;
    description: string;
    comments: string[]
}

export type NoteCreate = Omit<Note, \\"id\\">;
export type NoteChange =  Pick<Note, \\"id\\"> && Partial<NoteCreate>;

export interface Comment {
    id: string;
    title: string;
    description: string;
    noteComment: string
}

export type CommentCreate = Omit<Comment, \\"id\\">;
export type CommentChange =  Pick<Comment, \\"id\\"> && Partial<CommentCreate>;
"
`;

exports[`generate 2`] = `
"{
  \\"Note\\": {
    \\"name\\": \\"Note\\",
    \\"version\\": 1,
    \\"type\\": \\"object\\",
    \\"primaryKey\\": \\"id\\",
    \\"properties\\": {
      \\"id\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"id\\",
        \\"isRequired\\": true,
        \\"index\\": true,
        \\"primary\\": true
      },
      \\"title\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"title\\",
        \\"isRequired\\": true
      },
      \\"description\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"description\\",
        \\"isRequired\\": true
      },
      \\"comments\\": {
        \\"type\\": \\"string[]\\",
        \\"key\\": \\"comments\\",
        \\"isRequired\\": true
      }
    }
  },
  \\"Comment\\": {
    \\"name\\": \\"Comment\\",
    \\"version\\": 1,
    \\"type\\": \\"object\\",
    \\"primaryKey\\": \\"id\\",
    \\"properties\\": {
      \\"id\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"id\\",
        \\"isRequired\\": true,
        \\"index\\": true,
        \\"primary\\": true
      },
      \\"title\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"title\\",
        \\"isRequired\\": true
      },
      \\"description\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"description\\",
        \\"isRequired\\": true
      },
      \\"noteComment\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"noteComment\\",
        \\"isRequired\\": true
      }
    }
  }
}"
`;

exports[`generate 3`] = `
"import { ModelJsonSchema } from \\"offix-datastore\\";
import jsonSchema from \\"./schema.json\\";

type Schema<T = any> = {
    [P in keyof T]: ModelJsonSchema<any>
};

export const schema = jsonSchema as Schema;
"
`;
